### PR TITLE
UI Bug Fixes for Login & Signup

### LINKED ISSUE #98: Youssef's Last Bug Fixes
# Summary

1. **Updated minimum password requirement to 8 characters**
2. **Keyboard no longer pushes content up on login and signup pages**
3. **Error messages have preallocated space, so they do not mess with the layout of the login and signup pages**
4. **Small adjustments to padding and making sure Forgot Password is right-justified**

### PR DESCRIPTION
**Pull request by youssefctrl**
Close #98 
Simulator Screenshot - iPhone 16 Pro - 2024-12-12 at 22 17 26
Simulator Screenshot - iPhone 16 Pro - 2024-12-12 at 22 17 31
Simulator Screenshot - iPhone 16 Pro - 2024-12-12 at 22 17 36

### CODE REVIEW

File: FlatMate/View/SignupView.swift [hunk 1]
```diff
@@ -119,8 +124,8 @@ struct SignupView: View {
             return false
         }
         
-        if password.count < 12 {
-            errorMessage = "Password must be at least 12 characters long."
+        if password.count < 8 {
+            errorMessage = "Password must be at least 8 characters long."
```
Comment (Anidion) [inline FlatMate/View/SignupView.swift:127]: Do we validate this change on Firebase's end? If not we're good to merge

**Review by Anidion — APPROVED** [review]
Looks fantastic, nice work

File: FlatMate/Components/InputView.swift
Comment (Anidion) [inline FlatMate/Components/InputView.swift:1]: Re-added the negative bottom padding to make this a bit tighter

File: FlatMate/View/SignupView.swift (hunk 1 above)
Comment (youssefctrl) [reply FlatMate/View/SignupView.swift:127]: It looks like we only validate on the client-side before sending any data to Firebase, so we should be good to merge!

File: FlatMate/Components/InputView.swift
Comment (youssefctrl) [reply FlatMate/Components/InputView.swift:1]: Perfect, thank you!

**Pull request merged**